alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java

```diff
@@ -2656,7 +2656,7 @@ public void indexTransaction(Transaction info, boolean overwrite) throws IOExcep
 
     private void indexNonShardCascade(NodeMetaData nodeMetaData) throws IOException
     {
-        canUpdate();
```

Comment:
How do we know that it's safe to update? If for example the metadata tracker throws an exception and a roll back is required we should not commit the cascade update as it could be related to the metadata thats been rolled back.

Reply:
The indexNonShardCascade is called by the metadata tracker and the changes are not commited here. Notice that the writeLock of metadata tracker is acquired here.

The commit Tracker in order to commit the changes needs to acquire the writeLock of metadata tracker. In case of rollback, the changes are not committed by the commit tracker  and the rollback is correctly performed.

The canUpdate call seems to be useless here.